beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -55,39 +55,145 @@ func (s *Server) ListAttestations(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	attestations = append(attestations, unaggAtts...)
+
 	isEmptyReq := rawSlot == "" && rawCommitteeIndex == ""
+	bothDefined := rawSlot != "" && rawCommitteeIndex != ""
+	var attsData json.RawMessage
+
 	if isEmptyReq {
 		allAtts := make([]*structs.Attestation, len(attestations))
 		for i, att := range attestations {
 			a, ok := att.(*eth.Attestation)
-			if ok {
-				allAtts[i] = structs.AttFromConsensus(a)
-			} else {
-				httputil.HandleError(w, fmt.Sprintf("unable to convert attestations of type %T", att), http.StatusInternalServerError)
+			if !ok {
+				httputil.HandleError(w, fmt.Sprintf("Unable to convert attestations of type %T", att), http.StatusInternalServerError)
 				return
 			}
+			allAtts[i] = structs.AttFromConsensus(a)
+		}
+		attsData, err = json.Marshal(allAtts)
+		if err != nil {
+			httputil.HandleError(w, "Could not marshal attestations: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+	} else {
+		filteredAtts := make([]*structs.Attestation, 0, len(attestations))
+		for _, att := range attestations {
+			committeeIndexMatch := rawCommitteeIndex != "" && att.GetData().CommitteeIndex == primitives.CommitteeIndex(committeeIndex)
+			slotMatch := rawSlot != "" && att.GetData().Slot == primitives.Slot(slot)
+			shouldAppend := (bothDefined && committeeIndexMatch && slotMatch) || (!bothDefined && (committeeIndexMatch || slotMatch))
+			if shouldAppend {
+				a, ok := att.(*eth.Attestation)
+				if !ok {
+					httputil.HandleError(w, fmt.Sprintf("Unable to convert attestations of type %T", att), http.StatusInternalServerError)
+					return
+				}
+				filteredAtts = append(filteredAtts, structs.AttFromConsensus(a))
+			}
 		}
-		httputil.WriteJson(w, &structs.ListAttestationsResponse{Data: allAtts})
+		attsData, err = json.Marshal(filteredAtts)
+		if err != nil {
+			httputil.HandleError(w, "Could not marshal attestations: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+	httputil.WriteJson(w, &structs.ListAttestationsResponse{
+		Data: attsData,
+	})
+}
+
+// ListAttestationsV2 retrieves attestations known by the node but
+// not necessarily incorporated into any block. Allows filtering by committee index or slot.
+func (s *Server) ListAttestationsV2(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "beacon.ListAttestationsV2")
+	defer span.End()
+
+	rawSlot, slot, ok := shared.UintFromQuery(w, r, "slot", false)
```

Comment:
I know we have this pattern of !ok for listattestations but I don't know why we wouldn't error here ... wouldn't this be bad request? ( might need to double check the specs)

Reply:
The slot isn't a required field so, I am not sure why we would error here

Reply:
@james-prysm We write errors to `w` inside the function